crates/compiler/gen_llvm/src/llvm/build.rs

```diff
@@ -2964,136 +2964,61 @@ fn list_literal<'a, 'ctx>(
     let list_length = elems.len();
     let list_length_intval = env.ptr_int().const_int(list_length as _, false);
 
-    // TODO re-enable, currently causes morphic segfaults because it tries to update
-    // constants in-place...
-    // if element_type.is_int_type() {
-    if false {
+    let is_all_constant = elems.iter().all(|e| e.is_literal());
+
+    if is_all_constant {
         let element_type = element_type.into_int_type();
```

Comment:
You all through all constants, but still coerce the type into an `int_type`, that seems wrong? What would that do with a list, record, string, or tag? Can we safely coerce those into int types and expect it to work?

Reply:
yes, this needs to updated to just be for ints. I didn't have time to make that change when i made this pr.